### DISCUSSION TITLE
XPath Query Encoding

### DISCUSSION THREAD

**Discussion by miller9904** [searxng/searxng#1924] [Q&A] [answered]
I am currently working on a custom XPath engine for my local county library.  My issue is that when my search query is encoded, spaces are encoded as a '+', which the website does not understand to be a space.  This causes the website to return no results.  Is there a way to change the replacement character from '+' to '%20'?

Here is the search_url: https://encore.greenvillelibrary.org/iii/encore/search/C__S{query}__Ff%3Afacetcollections%3A36%3A36%3ASouth%20Carolina%20Collection%3A%3A__P{pageno}__Orightresult__U__X0?lang=eng&suite=cobalt

**Comment by allendema** [comment, marked as answer]
If you create a copy of the xpath engine e.x custom_xpath.py: edit https://github.com/searxng/searxng/blob/67eea86b35a1f5b9f5f17e0438332f15e852a22e/searx/engines/xpath.py#L21 and https://github.com/searxng/searxng/blob/67eea86b35a1f5b9f5f17e0438332f15e852a22e/searx/engines/xpath.py#L176

To use ```quote_plus``` instead of urlencode.

For more see:
https://christophergs.com/python/2016/12/03/python-urllib-parse/
